particle-iot/device-os
# [gen3] bootloader: workaround incompatibility that required intermediate update through 1.1.0

### Problem

Gen 3 Device OS release < 1.1.0 had an incorrect IRQ priority setting for POWER_CLOCK_IRQn (fixed in https://github.com/particle-iot/device-os/commit/040210e1e4e7f793f62b8cd2bb3e740ffc170d82) which caused SoftDevice to panic. A change introduced in https://github.com/particle-iot/device-os/pull/1948 to disable this IRQ before jumping from the bootloader into the system part triggered this problem.

In order to resolve this problem we've introduced a dependency on 1.1.0 in the bootloader, so that SMH could first update system part to an unaffected version.

### Solution

There is an option of simply masking any POWER_CLOCK_IRQn interrupts to achieve the same effect as it just being disabled.

This removes the dependency from the bootloader and just removes one more thing we have to worry about.

### Steps to Test

- Make sure that bootloader built out of this branch works correctly on Argon, Boron and B SoM with releases < 1.1.0.
- Make sure that 'Steps to test' from https://github.com/particle-iot/device-os/pull/1948 do not show any problems

### Example App

N/A

### References

- https://github.com/particle-iot/device-os/pull/1948
- https://github.com/particle-iot/device-os/commit/040210e1e4e7f793f62b8cd2bb3e740ffc170d82

---

### Completeness
